cocos/renderer/backend/opengl/ProgramGL.cpp

```diff
@@ -17,6 +20,21 @@ ProgramGL::ProgramGL(const std::string& vertexShader, const std::string& fragmen
     CC_SAFE_RETAIN(_fragmentShaderModule);
     compileProgram();
     computeUniformInfos();
+
+#if CC_ENABLE_CACHE_TEXTURE_DATA
+    for(const auto& uniform: _uniformInfos)
+    {
+        UniformLocation uniformLocation;
+        uniformLocation.location = glGetUniformLocation(_program, uniform.first.c_str());
+        _originalUniformLocations[uniform.first] = uniformLocation;
+        _uniformLocationMap[uniform.second.location] = uniform.second.location;
+    }
+
+    _backToForegroundListener = EventListenerCustom::create(EVENT_RENDERER_RECREATED, [this](EventCustom*){
```

Comment:
where to remove the listener?

Reply:
I forget to remove the listener, and it will be removed in the destructor.